compiler/rustc_typeck/src/check/compare_method.rs

```diff
@@ -64,10 +68,7 @@ pub(crate) fn compare_impl_method<'tcx>(
         return;
     }
 
-    if let Err(_) = compare_predicate_entailment(tcx, impl_m, impl_m_span, trait_m, impl_trait_ref)
-    {
-        return;
-    }
+    tcx.ensure().compare_predicates_and_trait_impl_trait_tys(impl_m.def_id);
```

Comment:
I think the previous pattern existed, even if simplifyable, so that it is easy to add something after it

Reply:
I don't think something's been added here in years. I could perhaps make the query return a `Result` so we could preserve this. Do you think that's necessary?

Reply:
By having making just the last call a query, we could get to a state where it is called without the earlier tests to succeed. Can this be an issue?

Reply:
Perhaps we would give nonsense errors that are avoided by these early returns, though I don't think this is a soundness risk or anything. Maybe it would be better to pull this whole "compare method" logic into one query so we short-circuit if any of those other checks fail.